db/mongo/mongodb.go

```diff
@@ -96,36 +99,45 @@ func (m *mongoClient) RemoveTask(taskID string) error {
 
 // ListTasks removes the tasks from mongo db
 func (m *mongoClient) ListTasks() ([]*models.Task, error) {
+	var task *models.Task
+
 	collection := m.conn.Database(viper.GetString(config.DbName)).Collection(taskCollection)
-	// Create an empty slice to store the retrieved tasks
-	tasks := []*models.Task{}
 
-	// Create an empty filter since we want to retrieve all tasks
+	tasks := []*models.Task{}
 	filter := bson.M{}
-	// Execute the find operation on the collection
-	cursor, err := collection.Find(context.Background(), filter)
+
+	cursor, _ := collection.Find(context.Background(), filter)
```

Comment:
why are we sending context.Background() here.
we should be initialising the context in the handlers and use the same context to call Service and DB Layers.